src/main/java/com/puppycrawl/tools/checkstyle/api/AstFilter.java

```diff
@@ -0,0 +1,34 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.api;
+
+/**
+ * An interface for filtering {@code TreeWalkerAuditEvent}.
+ *
+ * @author Timur Tibeyev.
+ */
+public interface AstFilter {
```

Comment:
> I dislike Ast slightly as Javadoc is a form of AST even though we call it Node. If Java upgrades to ANTLR4, will we rename these classes accordingly too and break previous compatibility? And how will we name them differently so not to confuse with Javadoc as both will be nodes?

Ok, lets name them as TreeWalkerFilter and TreeWalkerAuditEvent. 

@timurt 
Please, do.